test/connectionUI.test.ts

```diff
@@ -83,21 +89,42 @@ suite('Connection UI tests', () => {
 			quickPickItemType: CredentialsQuickPickItemType.Mru,
 			label: undefined
 		};
-		let mockConnection = { connectionString: 'test' };
-		prompter.setup(p => p.promptSingle(TypeMoq.It.isAny())).returns(() => Promise.resolve(item));
-		prompter.setup(p => p.prompt(TypeMoq.It.isAny(), true)).returns(() => Promise.resolve(mockConnection));
-		return connectionUI.promptForConnection().then(() => {
-			connectionStore.verify(c => c.getPickListItems(), TypeMoq.Times.once());
-			prompter.verify(p => p.promptSingle(TypeMoq.It.isAny()), TypeMoq.Times.once());
-		});
+
+		let quickPickMock = TypeMoq.Mock.ofType<vscode.QuickPick<IConnectionCredentialsQuickPickItem>>();
+		quickPickMock.setup(q => q.items);
+		quickPickMock.object.items = [item];
+		quickPickMock.setup(q => q.show());
+		quickPickMock.setup(q => q.onDidChangeSelection((e) => [item] = e));
+		quickPickMock.object.onDidChangeSelection((e) => [item] = e);
+		quickPickMock.setup(q => q.onDidHide(() => false));
+		vscodeWrapper.setup(v => v.createQuickPick()).returns(() => quickPickMock.object);
+
+		connectionUI.promptForConnection();
+		connectionStore.verify(c => c.getPickListItems(), TypeMoq.Times.once());
+		quickPickMock.verify(q => q.show(), TypeMoq.Times.once());
 	});
 
 	test('showConnections with recent but no selection', () => {
-		prompter.setup(p => p.promptSingle(TypeMoq.It.isAny())).returns(() => Promise.resolve(undefined));
-		return connectionUI.promptForConnection().then(() => {
-			connectionStore.verify(c => c.getPickListItems(), TypeMoq.Times.once());
-			prompter.verify(p => p.promptSingle(TypeMoq.It.isAny()), TypeMoq.Times.once());
-		});
+		let testCreds = new ConnectionCredentials();
+		testCreds.connectionString = 'test';
+		let item: IConnectionCredentialsQuickPickItem = {
+			connectionCreds: testCreds,
+			quickPickItemType: CredentialsQuickPickItemType.Profile,
+			label: undefined
+		};
+		let quickPickMock = TypeMoq.Mock.ofType<vscode.QuickPick<IConnectionCredentialsQuickPickItem>>();
+		quickPickMock.setup(q => q.items);
+		quickPickMock.object.items = [item];
+		quickPickMock.setup(q => q.show());
+		quickPickMock.setup(q => q.onDidChangeSelection((e) => [] = e));
+		quickPickMock.object.onDidChangeSelection(undefined);
+		quickPickMock.setup(q => q.onDidHide(() => false));
+		vscodeWrapper.setup(v => v.createQuickPick()).returns(() => quickPickMock.object);
```

Comment:
these tests look like they have a lot of the same setup code. Can there be a helper function or beforeEach that does the common setup?